src/analysis/calculus/formal_multilinear_series.lean

```diff
@@ -164,3 +170,122 @@ lemma comp_formal_multilinear_series_apply'
+noncomputable def order (p : formal_multilinear_series 𝕜 E F) : ℕ :=
+Inf { n | p n ≠ 0 }
+
+@[simp] lemma order_zero : (0 : formal_multilinear_series 𝕜 E F).order = 0 := by simp [order]
+
+lemma ne_zero_of_order_ne_zero (hp : p.order ≠ 0) : p ≠ 0 :=
+λ h, by simpa [h] using hp
+
+lemma order_eq_find [decidable_pred (λ n, p n ≠ 0)] (hp : ∃ n, p n ≠ 0) :
+  p.order = nat.find hp :=
+by simp [order, Inf, hp]
+
+lemma order_eq_find' [decidable_pred (λ n, p n ≠ 0)] (hp : p ≠ 0) :
+  p.order = nat.find (formal_multilinear_series.ne_iff.mp hp) :=
+order_eq_find _
+
+lemma order_eq_zero_iff (hp : p ≠ 0) : p.order = 0 ↔ p 0 ≠ 0 :=
+begin
+  classical,
+  have : ∃ n, p n ≠ 0 := formal_multilinear_series.ne_iff.mp hp,
+  simp [order_eq_find this, hp]
+end
+
+lemma order_eq_zero_iff' : p.order = 0 ↔ p = 0 ∨ p 0 ≠ 0 :=
+by { by_cases h : p = 0; simp [h, order_eq_zero_iff] }
+
+lemma apply_order_ne_zero (hp : p ≠ 0) : p p.order ≠ 0 :=
+begin
+  classical,
+  let h := formal_multilinear_series.ne_iff.mp hp,
+  exact (order_eq_find h).symm ▸ nat.find_spec h
+end
+
+lemma apply_order_ne_zero' (hp : p.order ≠ 0) : p p.order ≠ 0 :=
+apply_order_ne_zero (ne_zero_of_order_ne_zero hp)
+
+lemma apply_eq_zero_of_lt_order (hp : n < p.order) : p n = 0 :=
+begin
+  by_cases p = 0,
+  { simp [h] },
+  { classical,
+    rw [order_eq_find' h] at hp,
+    simpa using nat.find_min _ hp }
+end
+
+end order
+
+section coef
+
+variables [nontrivially_normed_field 𝕜]
+  [normed_add_comm_group E] [normed_space 𝕜 E] {s : E}
+  {p : formal_multilinear_series 𝕜 𝕜 E} {f : 𝕜 → E}
+  {n : ℕ} {z z₀ : 𝕜} {y : fin n → 𝕜}
+
+open_locale big_operators
+
+/-- The `n`th coefficient of `p` when seen as a power series. -/
+def coeff (p : formal_multilinear_series 𝕜 𝕜 E) (n : ℕ) : E := p n 1
+
+lemma mk_pi_field_coeff_eq (p : formal_multilinear_series 𝕜 𝕜 E) (n : ℕ) :
+  continuous_multilinear_map.mk_pi_field 𝕜 (fin n) (p.coeff n) = p n :=
+(p n).mk_pi_field_apply_one_eq_self
+
+@[simp] lemma apply_eq_prod_smul_coeff : p n y = (∏ i, y i) • p.coeff n :=
+begin
+  convert (p n).to_multilinear_map.map_smul_univ y 1,
+  funext; simp only [pi.one_apply, algebra.id.smul_eq_mul, mul_one],
+end
+
+lemma coeff_eq_zero : p.coeff n = 0 ↔ p n = 0 :=
+by rw [← mk_pi_field_coeff_eq p, continuous_multilinear_map.mk_pi_field_eq_zero_iff]
+
+@[simp] lemma apply_eq_pow_smul_coeff : p n (λ _, z) = z ^ n • p.coeff n :=
+by simp
+
+@[simp] lemma norm_apply_eq_norm_coef : ∥p n∥ = ∥coeff p n∥ :=
+by rw [← mk_pi_field_coeff_eq p, continuous_multilinear_map.norm_mk_pi_field]
+
+end coef
+
+section fslope
+
+variables [nontrivially_normed_field 𝕜]
+  [normed_add_comm_group E] [normed_space 𝕜 E]
+  {p : formal_multilinear_series 𝕜 𝕜 E} {n : ℕ}
+
+/-- The formal counterpart of `dslope`, corresponding to the expansion of `(f z - f 0) / z`. If `f`
+has `p` as a power series, then `dslope f` has `fslope p` as a power series. -/
+noncomputable def fslope (p : formal_multilinear_series 𝕜 𝕜 E) : formal_multilinear_series 𝕜 𝕜 E :=
```

Comment:
This can be generalized to `formal_multilinear_series` from a normed algebra to a normed space, but I'm not sure there is any point in doing so, let's wait for Sébastien's opinion.